ODIN_II/SRC/OdinGridAnalyzer.cpp

```diff
@@ -0,0 +1,334 @@
+                   type->name, region_height, xspec.repeat_expr.c_str(), repeaty);
+        }
+
+        size_t x_end = 0;
+        for (size_t kx = 0; x_end < grid_width; ++kx) { //Repeat in x direction
+            size_t x_start = startx + kx * repeatx;
+            x_end = endx + kx * repeatx;
+
+            size_t y_end = 0;
+            for (size_t ky = 0; y_end < grid_height; ++ky) { //Repeat in y direction
+                size_t y_start = starty + ky * repeaty;
+                y_end = endy + ky * repeaty;
+
+                size_t x_max = std::min((int)x_end, grid_width - 1);
+                size_t y_max = std::min((int)y_end, grid_height - 1);
+
+                //Fill in the region
+                for (size_t x = x_start; x + (type->width - 1) <= x_max; x += incrx) {
+                    for (size_t y = y_start; y + (type->height - 1) <= y_max; y += incry) {
+						
+                        set_grid_block_type(grid_loc_def->priority, type, x, y, grid);
+                    }
+                }
+            }
+        }
+	}
+/**
+ *  First is type (char*), second is priority
+ */ 
+void OdinGridAnalyzer::set_grid_block_type(int priority,t_physical_tile_type* type, int x_root, int y_root,std::vector<std::vector<std::pair<char*,int>>>&  grid){
+	if (grid[x_root][y_root].second < priority){
+            grid[x_root][y_root].first = type->name;
+            grid[x_root][y_root].second = priority;
+
+	// This block is commented out, to simplify counting
+	// if this code is enabled, the hard block will occupy as
+	// much grid tiles as it should, however, a special counting
+	// procedure should be implemented. That being said,
+	// somebody might need that code in future, whereas
+	// it is not required at current state
+
+    // for (size_t x = x_root; x < x_root + type->width; ++x) {
+    //     size_t x_offset = x - x_root;
+    //     for (size_t y = y_root; y < y_root + type->height; ++y) {
+    //         size_t y_offset = y - y_root;
+    //         auto& grid_tile = grid[x][y];
+    //         grid[x][y].first = type->name;
+    //         grid[x][y].second = priority;
+    //     }
+    // }
+
+	}
+}
+
+int OdinGridAnalyzer::findDesiredBlockPriority(t_grid_def& layout,std::string& typeTag){
+	int result = -1;
+	std::vector<t_grid_loc_def>::iterator pointer;
+	for (pointer =  layout.loc_defs.begin();pointer<layout.loc_defs.end(); pointer++){
+		if (typeTag.compare(pointer->block_type) == 0){
+			result =  pointer->priority;
+		}
+	}
+	//TODO are there any unspecified priorities?
+	return result;
+}
+
+ssize_t OdinGridAnalyzer::findIndexOfAssociatedTileType(std::vector<t_physical_tile_type> tileTypes,std::string& typeTag){
+	ssize_t result = -1;
+	ssize_t iter = -1;
+	std::cout<< "The type tag is"<<typeTag<<std::endl;
+	std::vector<t_physical_tile_type>::iterator pointer;
+
+	for (pointer =  tileTypes.begin();pointer< tileTypes.end(); pointer++){
```

Comment:
Use a foreach loop, the iterator one is only used if you are to delete anything from the vector